elfo-network/src/discovery/mod.rs

```diff
@@ -550,6 +584,37 @@ async fn recv_regular<M: Message>(socket: &mut Socket, idle_timeout: Duration) -
     })
 }
 
+#[cfg_attr(test, derive(Debug))]
+struct Diff<T> {
+    new: Vec<T>,
+    removed: fxhash::FxHashSet<T>,
+}
+
+fn diff_owned<'a, T, I>(old: fxhash::FxHashSet<T>, new: I) -> Diff<T>
```

Comment:
What about `Diff::new(old: impl IntoIterator, new: impl IntoIterator)`?

Reply:
IMO better name would be `Diff::make`, and, about making `old` IntoIterator instead: I think showing `HashSet` in constructor is better, since it highlights that calling this API may be not that efficient (first you need to build hash set, then operate on it), gives caller opportunity to optimize diffs (by constructing hashset from ground up, for example). While these points are about optimization and not so important here, however, leaving opportunities here is easier than not leaving, so I decided to design API this way.